src/Core/Addon/Module/ModuleInterface.php

```diff
@@ -81,4 +81,11 @@ public function onReset(): bool;
      * @return bool true for success
      */
     public function onUpgrade($version): bool;
+
+    /**
+     * Return true if module is active
+     *
+     * @return bool
+     */
+    public function isActive();
```

Comment:
```suggestion
    public function isActive(): bool;
```

Reply:
@Progi1984 i first already add this signature to the method but Phpstan fail as the return of src/Core/Addon/Module/ModuleInterface:isActive is not signed.
Should i update this class too ?

Reply:
Yes. In all cases, you create a BC Break like you define new methods in interfaces. In this case, it's better to typehint methods.

Reply:
If until today all modules had to extend `Module` base class and can't implement `ModuleInterface` as an alternative implementation, then there is no BC break from the new method in the interface.
But typing the method will result create a BC break.
Question: do all modules have to extend `Module` base class because it is the expected type instead of the interface ?

Reply:
> Question: do all modules have to extend Module base class because it is the expected type instead of the interface ?

Yes, as stated in the developer documentation => https://devdocs.prestashop.com/8/modules/creation/tutorial/

I guess this is because Modules not only need to provide some methods, there are also properties that have to be defined. The documentation mentions them.

Reply:
Possibly yes... a part of my soul is dying each time I meet a public prop :p